Tests/Pcap++Test/Tests/TcpReassemblyTests.cpp

```diff
@@ -1264,3 +1264,31 @@ PTF_TEST_CASE(TestTcpReassemblyTimeStamps)
 	packetStream.clear();
 	tcpReassemblyResults.clear();
 } // TestTcpReassemblyTimeStamps
+
+PTF_TEST_CASE(TestTcpReassemblyFinishReset)
+{
+	TcpReassemblyMultipleConnStats results;
+	std::string errMsg;
+
+	std::vector<pcpp::RawPacket> packetStream;
+	PTF_ASSERT_TRUE(readPcapIntoPacketVec("PcapExamples/one_tcp_stream_fin_rst_close_packet.pcap", packetStream, errMsg));
+
+	pcpp::TcpReassemblyConfiguration config(true, 2, 1);
+	pcpp::TcpReassembly tcpReassembly(tcpReassemblyMsgReadyCallback, &results, tcpReassemblyConnectionStartCallback,
+	                                  tcpReassemblyConnectionEndCallback, config);
```

Comment:
I think a better approach would be to call `tcpReassemblyTest(packetStream, tcpReassemblyResults, true, false)` and check the results:
- Verify there is only 1 connection
- Verify that `connectionsStarted == true`, `connectionsEnded == false`, `connectionsEndedManually == false`
- Verify that the number of packets from each side is what we expect